Add metadata command

amberstarlight opened the pull request.

amberstarlight committed:
Add info command with metadata

mini-ninja-64 commented on commands/info.js:
```diff
@@ -0,0 +1,41 @@
+const ms = require('ms');
+const wait = require('util').promisify(setTimeout);
+
+const gitSha =
+	process.env.GIT_SHA || '0ed46333853d1aaab845c769a88845b4fd7b9dde';
+const commitMessage =
+	process.env.COMMIT_MESSAGE ||
+	`Remove Arm images, we no longer need these (#40)
+
+* Remove Arm images, we no longer need these
+* Remove strings that aren't needed`;
```
Do we need default data here?
if we do for testing with cutiebot-dev maybe it would be clearer to instead have data that is clearly used for testing :thinking:

amberstarlight replied:
Changed in [37d4691](https://github.com/Cutie-Club/cutiebot/pull/41/commits/37d46915f81d72dcda3a031270b01dfca332c0cb)

mini-ninja-64 commented on commands/info.js:
```diff
@@ -0,0 +1,41 @@
+module.exports = {
+	data: new SlashCommandBuilder()
+		.setName('info')
+		.setDescription('Prints metadata.'),
```
should we change metadata to info for consistency

amberstarlight replied:
Changed in [cb69533](https://github.com/Cutie-Club/cutiebot/pull/41/commits/cb69533ba9fee63c62f8560c18779f4cd591488e)

mini-ninja-64 commented on commands/info.js:
```diff
@@ -0,0 +1,41 @@
+* Remove Arm images, we no longer need these
+* Remove strings that aren't needed`;
+
+const uptime = async () => {
+	if (!process.uptime) await wait(1000);
+	return ms(Math.floor(process.uptime() * 1000), { long: true });
+};
```
```suggestion
const uptime = () => {
	return ms(process.uptime() * 1000), { long: true });
};
```
Does it need to be async or rounded, can't we just do something simple like this?

amberstarlight replied:
Yes we can skip all of this. Changed in [6453e23](https://github.com/Cutie-Club/cutiebot/pull/41/commits/6453e23adfc7c8ff47d568d5376212587a087a5e)

mini-ninja-64 commented on commands/info.js:
```diff
@@ -0,0 +1,41 @@
+						)}\`](https://github.com/Cutie-Club/cutiebot/commit/${gitSha})`
+					)
+					.addField('Commit message', `\`\`\`text\n${commitMessage}\n\`\`\``)
+					.addField('Uptime', `${await uptime()}`),
```
uptime returns a string so i would give it a name like `uptimeString` so it doesnt compete with `process.uptime`

i dont think we need to wrap it as a string, since it returns a string already

amberstarlight replied:
Changed in [6453e23](https://github.com/Cutie-Club/cutiebot/pull/41/commits/6453e23adfc7c8ff47d568d5376212587a087a5e)

mini-ninja-64 reviewed (changes requested):
Some thoughts :pleading_face:

amberstarlight committed:
Refine uptimeString function

amberstarlight committed:
Rename info to metadata for consistency

amberstarlight committed:
Remove default git data
- Improve interaction error UX
- Set required environment variables in dev:start command

amberstarlight committed:
Add check-fmt command so that prettier failures cause ci to exit

amberstarlight committed:
Remove unused const

amberstarlight committed:
Improve error message embed

amberstarlight committed:
Use newline escape characters instead

mini-ninja-64 reviewed (approved):
<img src="https://media4.giphy.com/media/vhsNmFjuN4WDS/giphy.gif"/>
